### PR TITLE
[Placement Group] Make PlacementGroupID generate from JobID

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?
Make `PlacementGroupId`  generate from `JobId`.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
I think ray anyway doesn't work with fork()? Can you elaborate the exception case?

File: src/ray/common/id.h [hunk 1]
```diff
@@ -308,17 +308,36 @@ class ObjectID : public BaseID<ObjectID> {
 };
 
 class PlacementGroupID : public BaseID<PlacementGroupID> {
+ private:
+  static constexpr size_t kUniqueBytesLength = 14;
+
  public:
-  static constexpr size_t kLength = 16;
+  /// Length of `ActorID` in bytes.
```
Comment (ffbin) [inline src/ray/common/id.h:315]:
```suggestion
  /// Length of `PlacementGroupID` in bytes.
```

File: src/ray/common/id.h [hunk 2]
```diff
@@ -308,17 +308,36 @@ class ObjectID : public BaseID<ObjectID> {
...
+  static constexpr size_t kLength = kUniqueBytesLength + JobID::kLength;
 
   /// Size of `PlacementGroupID` in bytes.
   ///
   /// \return Size of `PlacementGroupID` in bytes.
   static size_t Size() { return kLength; }
 
+  /// Creates an `PlacementGroupID` by hashing the given information.
```
Comment (ffbin) [inline src/ray/common/id.h:323]:
```suggestion
  /// Creates a `PlacementGroupID` by hashing the given information.
```

**Comment by ffbin** [comment]
Hey @clay4444 Why `PlacementGroupID::Of(JobID)` will not generate duplicate IDs after a fork() call?

**Comment by clay4444** [comment]
> Hey @clay4444 Why `PlacementGroupID::Of(JobID)` will not generate duplicate IDs after a fork() call?

yeah, the reason why we make the `FromRandom` deleted is we want `PlacementGroupId` generate from `JobId` instead of from random, I have changed the comment.  cc @rkooo567

File: src/ray/common/id.h (hunk 1 above)
Comment (clay4444) [reply src/ray/common/id.h]: updated

**Comment by rkooo567** [comment]
Still didn't get answer for this question?

> I think ray anyway doesn't work with fork()? Can you elaborate the exception case?

Ray doesn't work with fork anyway. What's the use case, and adding this code will fix the whole issue?

**Review by rkooo567 — CHANGES_REQUESTED** [review]
So, there are 2 things.

1. Ray is not working with fork anyway. If we'd like to introduce this change, we should also add a python test that tests if it works with fork().
2. To do 1, we need a solid use case. There's no reason to fix this for a partial support without a use case.

**Comment by clay4444** [comment]
close temporarily.

**Pull request closed without merging**